library/mt_interfaces.py

```diff
@@ -64,12 +64,19 @@ def main():
       username=dict(required=True),
       password=dict(required=True, no_log=True),
       settings=dict(required=True, type='dict'),
-      parameter = dict(
-          required  = True,
-          choices   = ['ethernet', 'vlan', 'ovpn-client'],
-          type      = 'str'
+      parameter=dict(
+          required=True,
+          choices=[
```

Comment:
I see what you're doing here with the 'parameter' being the api endpoint, and everything else being passed in as settings.  This is probably a good pattern going forward, especially since mikrotik's api can and most likely will change.